### PR TITLE
Test guice setup and fix

### PR DESCRIPTION
**Pull request by christophsturm**
@sqrrm's guice fix and a unit test that checks seednode's guice setup. superseedes #3041

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -404,6 +404,8 @@ configure(project(':seednode')) {
         runtime "org.bouncycastle:bcprov-jdk15on:$bcVersion"
         compileOnly "org.projectlombok:lombok:$lombokVersion"
         annotationProcessor "org.projectlombok:lombok:$lombokVersion"
+        testCompile "org.springframework:spring-test:$springVersion"
+
```
Comment (sqrrm) [inline build.gradle]:
```suggestion
```

**Review by sqrrm — APPROVED** [review]
ACK

File: common/src/main/java/bisq/common/crypto/KeyStorage.java [hunk 2]
```diff
@@ -97,6 +97,7 @@ public String toString() {
 
     @Inject
     public KeyStorage(@Named(KEY_STORAGE_DIR) File storageDir) {
+        storageDir.mkdirs();
```
Comment (christophsturm) [inline common/src/main/java/bisq/common/crypto/KeyStorage.java:100]: the CI error came from a missing key directory on CI. I now changed the keystorage to create the directory if its missing. not sure if that has any downsides.

**Comment by christophsturm** [comment]
note that i had to bump guice to get a useful error message. I removed the guice bump again from this pr but we really should consider updating guice.

**Review by freimair — APPROVED** [review]
Ack

**Comment by christophsturm** [comment]
can we get this merged?

**Review by sqrrm — APPROVED** [review]
utACK

**Pull request merged**